components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3504,4 +3509,63 @@ protected APIManagerConfiguration getAPIManagerConfiguration() {
                 .getAPIManagerConfiguration();
     }
 
+    /**
+     * To get the query to retrieve user role list query based on current role list.
+     *
+     * @return the query with user role list.
+     * @throws APIManagementException API Management Exception.
+     */
+    private String getUserRoleListQuery() throws APIManagementException {
+        StringBuilder rolesQuery = new StringBuilder();
+        rolesQuery.append('(');
+        rolesQuery.append(APIConstants.NULL_USER_ROLE_LIST);
+        String[] userRoles = APIUtil.getListOfRoles(username, true);
+        if (userRoles != null) {
+            for (String userRole : userRoles) {
+                rolesQuery.append(" OR ");
+                rolesQuery.append(ClientUtils.escapeQueryChars(userRole.toLowerCase()));
+            }
+        }
+        rolesQuery.append(")");
```

Comment:
I believe it would be best to keep these strings as constants. Shall we add them as constants?